script.js

```diff
@@ -1,4 +1,340 @@
+// Variables to store player 1 and player 2's scores:
+var P1TotalScore = 0;
+var P2TotalScore = 0;
+
+// To track game modes & statuses
+var MODE_CHOOSE_GAME_MODE = `WAITING_FOR_USER_TO_CHOOSE_GAME_MODE`;
+var EITHER_HIGHEST_OR_LOWEST_WINS = `EITHER HIGHEST OR LOWEST WINS`;
+var MODE_HIGHER_NUMBER = `higher number`;
+var MODE_LOWER_NUMBER = `lower number`;
+var MODE_PLAYER_1_ROLL_DICE = `WAITING_FOR_PLAYER_1_TO_ROLL_DICE`;
+var MODE_PLAYER_1_PICK_ORDER = `WAITING_FOR_PLAYER_1_TO_PICK_DICE_ORDER`;
+var MODE_PLAYER_2_ROLL_DICE = `WAITING_FOR_PLAYER_2_TO_ROLL_DICE`;
+var MODE_PLAYER_2_PICK_ORDER = `WAITING_FOR_PLAYER_2_TO_PICK_DICE_ORDER`;
+var MODE_WINNER = `SHOW_WINNER`;
+
+//Current game mode
+var mode = MODE_CHOOSE_GAME_MODE;
+var theModeUserInput = EITHER_HIGHEST_OR_LOWEST_WINS;
+
+// Function for leaderboard for Mode: Higher number wins:
+var leaderBoardHigher = function () {
+  if (Number(P1TotalScore) > Number(P2TotalScore)) {
+    myOutputValue = `Player 1<br>Player 2`;
+  } else if (Number(P2TotalScore) > Number(P1TotalScore)) {
+    myOutputValue = `Player 2<br>Player 1`;
+  } else if (Number(P1TotalScore) == Number(P2TotalScore)) {
+    myOutputValue = `Draw`;
+  }
+  return myOutputValue;
+};
+
+// Function for leaderboard for Mode: Lower number wins:
+var leaderBoardLower = function () {
+  if (Number(P1TotalScore) < Number(P2TotalScore)) {
+    myOutputValue = `Player 1<br>Player 2`;
+  } else if (Number(P2TotalScore) < Number(P1TotalScore)) {
+    myOutputValue = `Player 2<br>Player 1`;
+  } else if (Number(P1TotalScore) == Number(P2TotalScore)) {
+    myOutputValue = `Draw`;
+  }
+  return myOutputValue;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var P1RandomDiceRoll1 = diceRoll();
+  var P1RandomDiceRoll2 = diceRoll();
+  var leaderBoardRanksHigher = leaderBoardHigher();
+  var leaderBoardRanksLower = leaderBoardLower();
+
+  // User to choose game mode
+  if (mode == MODE_CHOOSE_GAME_MODE) {
+    console.log(mode);
+    var myOutputValue = `Hello! Welcome to the game of BEAT THAT.<br><br>Please select your game mode preference.<br><br>Choose either "higher number" or "lower number". <br><br>Good luck!!`;
+    mode = MODE_PLAYER_1_ROLL_DICE;
+    console.log(mode);
+  } else if (input == `higher number`) {
```

Comment:
Good use of `else if`  to 1. assert control over code flow, and 2. highlight to readers (like myself) that there is a mutually exclusive relationship between these blocks of code (i.e. only either one of them can be true).